### PR TITLE
Implement show/hide code toggling functionality on component pages

### PR DESCRIPTION
**Pull request by contolini**
Look a thing that does a thing.

code-toggle

### CODE REVIEW

**Comment by cfarm** [comment]
reviewing now

File: docs/assets/css/toggle-code-button.less [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Button to show/hide code snippets and specs on component pages
+
+.hidden {
```
Comment (cfarm) [inline docs/assets/css/toggle-code-button.less]: it looks like the JS is just adding and removing the `u-hidden` class, so is any of this CSS used?

File: docs/_includes/variation/code-snippet.html [hunk 2]
```diff
@@ -1,7 +1,9 @@
 <div class="variation-code-snippet">
     {{ code_snippet }}
-    <h5 class="u-mt15">HTML</h5>
-    <div class="source-code" data-source-code>
-        <pre><code class="language-html">{{ code_snippet_raw }}</code></pre>
+    <div data-toggle-code>
+        <h5 class="u-mt15">HTML</h5>
```
Comment (cfarm) [inline docs/_includes/variation/code-snippet.html]: if you move the `u-mt15` class to the `variation-code-snippet` div above the margin whitespace will still be present when the code is hidden, which is nice.

File: docs/assets/css/toggle-code-button.less (hunk 1 above)
Comment (contolini) [reply docs/assets/css/toggle-code-button.less]: Yeah it's a mixin used on L11 and L15 but now that all it does is `display: none` I'll just inline that CSS below.

**Comment by contolini** [comment]
@cfarm yeah I figure we can use web storage for that.

**Pull request merged**